compiler-rt/lib/xray/xray_interface.cpp

```diff
@@ -471,42 +518,123 @@ uint16_t __xray_register_event_type(
 }
 
 XRayPatchingStatus __xray_patch() XRAY_NEVER_INSTRUMENT {
-  return controlPatching(true);
+  XRayPatchingStatus CombinedStatus{SUCCESS};
```

Comment:
I would argue yes, since in this case doing nothing would be the expected behavior. 
However, in practice this case should never occur because the executable itself is always "loaded".

We could add an assertion to make this assumption explicit. What do you think?

Reply:
> We could add an assertion to make this assumption explicit. What do you think?

Yeah - good idea, if it's easily done.

Reply:
After looking at it again, I have refactored the affected functions a bit to reduce code duplication and clarify the returned status.
I no objects are loaded, the returned status is `NOT_INITIALIZED`.